docs/intro/README.md

```diff
@@ -10,4 +10,99 @@ It is based on two major principles:
 
 - **Capabilities:** The SDK is inspired by capabilities-based security, and informed by years of wrestling with blockchain state-machines. Most developers will need to access other 3rd party modules when building their own modules. Given that the Cosmos-SDK is an open framework, some of the modules may be malicious, which means there is a need for security principles to reason about inter-module interactions. These principles are based on object-capabilities. In practice, this means that instead of having each module keep an access control list for other modules, each module implements special objects called keepers that can be passed to other modules to grant a pre-defined set of capabilities. For example, if an instance of module A's keepers is passed to module B, the latter will be able to call a restricted set of module A's functions. The capabilities of each keeper are defined by the module's developer, and it's the developer's job to understand and audit the safety of foreign code from 3rd party modules based on the capabilities they are passing into each third party module. For a deeper look at capabilities, jump to [this section](./ocap.md).
 
-### Next, learn more about the [SDK Application Architecture](./sdk-app-architecture.md)
+## SDK Application Architecture
+
+### State machine 
+
+At its core, a blockchain is a [replicated deterministic state machine](https://en.wikipedia.org/wiki/State_machine_replication). 
+
+A state machine is a computer science concept whereby a machine can have multiple states, but only one at any given time. There is a `state`, which describes the current state of the system, and `transactions`, that trigger state transitions. 
+
+Given a state S and a transaction T, the state machine will return a new state S'. 
+
+```
++--------+                 +--------+
+|        |                 |        |
+|   S    +---------------->+   S'   |
+|        |    apply(T)     |        |
++--------+                 +--------+
```

Comment:
This (and the next) diagram are maybe a bit superfluous here. I think an clearly written description of determinism as it relates the state machine (without assigning letters ex. `T` or using diagrams) would be adequate.

Reply:
I can touch on determinism, but I push back on the idea that the diagrams are superfluous. Remember that all this is **intro** material. We can assume no prior knowledge from people reading this, and we need to make it as gentle as possible.

A lot of people discovering the SDK come from the Ethereum community and may not even know what a state machine is. Or they may know it, but they may not be accustomed to thinking of a blockchain as a replicated, deterministic state-machine. What is obvious to us is NOT to most people getting started with the SDK.

We will have more advanced and rigorous docs later. The intro docs are high-level and must be as easy to understand as possible.